sinks/cortex/cortex.go

```diff
@@ -235,19 +236,24 @@ func (s *CortexMetricSink) Flush(ctx context.Context, metrics []samplers.InterMe
 			if len(batch)%s.batchWriteSize == 0 {
 				err := s.writeMetrics(ctx, batch)
 				if err != nil {
-					return err
+					s.logger.Error(err)
+					droppedMetrics += len(batch)
+				} else {
+					flushedMetrics += len(batch)
 				}
 
-				flushedMetrics += len(batch)
 				batch = []samplers.InterMetric{}
+				return err
 			}
 
 			return nil
 		})
 
-		if err != nil {
-			s.logger.Error(err)
-			droppedMetrics += len(metrics) - flushedMetrics
+		// Any other type of error will happen inside of doIfNotDone
+		// so we only need ot handle context cancellations
```

Comment:
Safer to maintain old behavior on other types of errors, otherwise it ignores, right?

Reply:
Ask being, maintain the if err != nil after
I'd do suggested change, but we're in a meeting right now so not focusing ;)